### PR TITLE
Docs: mention compatibility with cakephp4

### PR DESCRIPTION
**Pull request by LordSimal**
It seems like sometimes people think authentication version 2 is only compatible with CakePHP 2.
I will also create a PR for authorization plugin as well.

### CODE REVIEW

File: docs/en/index.rst [hunk 1]
```diff
@@ -8,6 +8,8 @@ Project's ROOT directory (where the **composer.json** file is located)
 
     php composer.phar require "cakephp/authentication:^2.0"
 
+Version 2 of the Authentication Plugin is compatible with CakePHP 4.
```
[the remark this answers is not in the document]
Comment (markstory) [reply docs/en/index.rst:11]: If we want one authoritative source, I would put it in the docs with each plugin. The wiki is not what most developers are going to look at when they need information.

[the remark this answers is not in the document]
Comment (dereuromark) [reply docs/en/index.rst:11]: I disagree, the Wiki is easy to maintain as single source of truth instead of multiple branches.

**Comment by LordSimal** [comment]
Should we add a link to the wiki now or update that line accordingly when it is required?

**Comment by dereuromark** [comment]
Sure, similar to core we can link to details in wiki

**Pull request merged**